frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -988,6 +1126,35 @@ pub mod pallet {
 	#[pallet::getter(fn snapshot_metadata)]
 	pub type SnapshotMetadata<T: Config> = StorageValue<_, SolutionOrSnapshotSize>;
 
+	/// The next index to be assigned to an incoming signed submission.
+	///
+	/// We can't just use `SignedSubmissionIndices.len()`, because that's a bounded set; past its
+	/// capacity, it will simply saturate. We can't just iterate over `SignedSubmissionsMap`,
+	/// because iteration is slow. Instead, we store the value here.
+	#[pallet::storage]
+	pub(crate) type SignedSubmissionNextIndex<T: Config> = StorageValue<_, u32, ValueQuery>;
```

Comment:
AFAICT we are using this an unique identifier for each submission during a signed phase? If this is the case I think a note somewhere mentioning that we expect every submitted solution to have a unique index during the signed phase would help with understanding the `SignedSubmissions` impl.

Reply:
65d7af60a

Reply:
https://github.com/paritytech/substrate/pull/7910/commits/65d7af60ac8b5294c6ee00466edc44f33cced6b7